src/NuGet.Clients/NuGet.PackageManagement.UI/Options/PackageSourcesOptionsControl.cs

```diff
@@ -676,6 +665,36 @@ private static bool IsPathRootedSafe(string path)
             // Otherwise, Path.IsPathRooted() will throw an ArgumentException.
             return path.IndexOfAny(Path.GetInvalidPathChars()) == -1 && Path.IsPathRooted(path);
         }
+
+        private IReadOnlyList<PackageSource> GetAllPackageSources()
+        {
+            return NuGetUIThreadHelper.JoinableTaskFactory.Run(async () => { return await GetAllPackageSourcesAsync(); });
+        }
+
+        private async Task<IReadOnlyList<PackageSource>> GetAllPackageSourcesAsync()
+        {
+            var remoteBroker = await BrokeredServicesUtilities.GetRemoteServiceBroker();
+            var nuGetSettingsService = await remoteBroker.GetProxyAsync<INuGetSourceRepositoryService>(NuGetBrokeredServices.SourceRepositoryProviderService);
+            using (nuGetSettingsService as IDisposable)
+            {
+                return await nuGetSettingsService.GetAllPackageSourcesAsync(CancellationToken.None);
```

Comment:
Always check whether the brokered service proxy you get back is `null` and degrade gracefully (e.g. disable the control, display a modeless message within the UI of the control, etc.) when you can't get the proxy back.

Reply:
I thought I had read somewhere that this would never return null, we would either get a remote/local proxy but not null.  I will start checking for null now when I do these.

Reply:
You can assume that a call to `GetFullAccessServiceBroker` will never return null, but you must assume that querying for a service may result in a null value. Thanks for adding the null checks.
If you add `#nullable enable` in these files, you'll be able to tell where in these APIs that nulls may come back.